cmd/soroban-rpc/internal/daemon/daemon.go

```diff
@@ -144,12 +148,24 @@ func MustNew(cfg *config.Config) *Daemon {
 	if len(cfg.HistoryArchiveURLs) == 0 {
 		logger.Fatal("no history archives url were provided")
 	}
-	historyArchive, err := historyarchive.Connect(
-		cfg.HistoryArchiveURLs[0],
+
+	historyArchive, err := historyarchive.NewArchivePool(
+		cfg.HistoryArchiveURLs,
 		historyarchive.ArchiveOptions{
+			NetworkPassphrase:   cfg.NetworkPassphrase,
 			CheckpointFrequency: cfg.CheckpointFrequency,
+			ConnectOptions: storage.ConnectOptions{
+				Context:   context.Background(),
+				UserAgent: fmt.Sprintf("soroban-rpc/%s", config.Version)},
+			CacheConfig: historyarchive.CacheOptions{
```

Comment:
I don't think it's necessary to cache history archive requests in soroban-rpc because soroban-rpc does not have a state verification routine which repeatedly downloads history archive files.

The only time soroban-rpc downloads the history archive files is if it's ingesting from a completely empty DB. In that case, soroban-rpc will ingest all ledger entries from the most recent checkpoint ledger. Otherwise, soroban-rpc will only ingest Tx Meta emitted by captive core.

If we add this history archive cache, I think we will not end up caching any requests and end up using extra disk for files we don't need.

Reply:
ok, that makes sense, will re-work this slightly to use the new ArchivePool interfaces still but disable cache usage.

Reply:
@tamirms , i see the [captive core backend also creates it's own internal archive pool](https://github.com/stellar/go/blob/master/ingest/ledgerbackend/captive_core_backend.go#L176) from urls, and then rpc also creates it's own archive pool locally from same urls, looking at the captive core backend, it primarily uses the pool up front to get root has and a ledger header, so, not heavy usage, but i'm wondering if it's worth consolidating the archive pool instances and letting the captive core backend receive optional pool instance rather than creating another one, such as this:
https://github.com/stellar/go/compare/master...sreuland:cc_archivepool

do you think it's worthwhile?

Reply:
even more general could be providing an `ArchiveInterface` (since `ArchivePool` conforms to that) xor the list of URLs

Reply:
I re-worked this to not enable ha caching and not emit archive metrics since there is such a limited usage of them from rpc as called out, only about 2-3 requests at start time is about it between rpc and captive core backend. less code is best I think for now rather than trying to get fancy, focused on getting the user agent config'd.

Reply:
@sreuland sounds good!